nvflare/private/fed/server/fed_server.py

```diff
@@ -596,28 +599,60 @@ def AuxCommunicate(self, request, context):
 
     def _aux_communicate(self, fl_ctx, shareable, shared_fl_context, topic):
         try:
-            with self.engine.lock:
-                job_id = shared_fl_context.get_prop(FLContextKey.CURRENT_RUN)
-                command_conn = self.engine.get_command_conn(str(job_id))
+            command_shareable = Shareable()
+            command_shareable.set_header(ServerCommandKey.PEER_FL_CONTEXT, shared_fl_context)
+            command_shareable.set_header(ServerCommandKey.TOPIC, topic)
+            command_shareable.set_header(ServerCommandKey.SHAREABLE, shareable)
+            data = {
+                ServerCommandKey.COMMAND: ServerCommandNames.AUX_COMMUNICATE,
+                ServerCommandKey.DATA: command_shareable,
+            }
+            if shareable.get_header(CollectiveCommShareableHeader.IS_COLLECTIVE_AUX, False):
+                with self.engine.lock:
```

Comment:
The COLLECTIVE_CONNECTION_WAITER has a timeout waiter wait for all the clients calling the AuxCommunicate, which will essentially block all the other server engine task execution because of the self.engine.lock. With a potential large self._collective_comm_timeout, it might cause the system issue?

Reply:
I only use ``with self.engine.lock`` to get the command waiter.

You can see L617 to L625 is NOT inside this with block.

So it does not block other server engine task execution.

Reply:
that's correct.